src/product_config_utils.rs

```diff
@@ -1107,15 +1114,19 @@ mod tests {
         let role_group_2 = "role_group_2";
         let file_name = "foo.bar";
 
-        let roles: HashMap<String, (Vec<PropertyNameKind>, Role<Box<TestConfig>>)> = collection! {
+        #[allow(clippy::type_complexity)]
+        let roles: HashMap<
+            String,
+            (Vec<PropertyNameKind>, Role<Box<TestConfig>, TestRoleConfig>),
+        > = collection! {
```

Comment:
This isn't pretty ... We should do something about this crazy config mechanism

Reply:
Agreed! But I would consider this to be out of scope of this PR